src/energyplus/Test/UnitarySystemPerformanceMultispeed_GTest.cpp

```diff
@@ -0,0 +1,435 @@
+
+    AirLoopHVAC airLoop(m);
+    Node supplyOutletNode = airLoop.supplyOutletNode();
+    unitary.addToNode(supplyOutletNode);
+
+    UnitarySystemPerformanceMultispeed perf(m);
+    perf.setName("US Perf Multispeed");
+    EXPECT_TRUE(perf.addSupplyAirflowRatioField(0.1, 0.2));
+    EXPECT_TRUE(perf.addSupplyAirflowRatioField(SupplyAirflowRatioField::fromCoolingRatio(0.3)));
+    EXPECT_TRUE(unitary.setDesignSpecificationMultispeedObject(perf));
+
+    ForwardTranslator ft;
+    Workspace w = ft.translateModel(m);
+
+    WorkspaceObjectVector idf_unitarys(w.getObjectsByType(IddObjectType::AirLoopHVAC_UnitarySystem));
+    EXPECT_EQ(1, idf_unitarys.size());
+    WorkspaceObject& idf_unitary = idf_unitarys.front();
+
+    EXPECT_EQ("UnitarySystemPerformance:Multispeed",
+              idf_unitary.getString(AirLoopHVAC_UnitarySystemFields::DesignSpecificationMultispeedObjectType).get());
+    EXPECT_EQ("US Perf Multispeed", idf_unitary.getString(AirLoopHVAC_UnitarySystemFields::DesignSpecificationMultispeedObjectName).get());
+
+    WorkspaceObjectVector idf_perfs(w.getObjectsByType(IddObjectType::UnitarySystemPerformance_Multispeed));
+    EXPECT_EQ(1, idf_perfs.size());
+    WorkspaceObject& idf_perf = idf_perfs.front();
+
+    EXPECT_EQ(1, idf_perf.getInt(UnitarySystemPerformance_MultispeedFields::NumberofSpeedsforHeating).get());
+    EXPECT_EQ(2, idf_perf.getInt(UnitarySystemPerformance_MultispeedFields::NumberofSpeedsforCooling).get());
+    EXPECT_EQ("No", idf_perf.getString(UnitarySystemPerformance_MultispeedFields::SingleModeOperation).get());
+    EXPECT_EQ(1.0, idf_perf.getDouble(UnitarySystemPerformance_MultispeedFields::NoLoadSupplyAirFlowRateRatio).get());
+
+    ASSERT_EQ(2u, idf_perf.numExtensibleGroups());
+
+    auto egs = idf_perf.extensibleGroups();
+
+    IdfExtensibleGroup eg1 = egs[0];
+    ASSERT_TRUE(eg1.getDouble(UnitarySystemPerformance_MultispeedExtensibleFields::HeatingSpeedSupplyAirFlowRatio));
+    EXPECT_EQ(0.1, eg1.getDouble(UnitarySystemPerformance_MultispeedExtensibleFields::HeatingSpeedSupplyAirFlowRatio).get());
+    ASSERT_TRUE(eg1.getDouble(UnitarySystemPerformance_MultispeedExtensibleFields::CoolingSpeedSupplyAirFlowRatio));
+    EXPECT_EQ(0.2, eg1.getDouble(UnitarySystemPerformance_MultispeedExtensibleFields::CoolingSpeedSupplyAirFlowRatio).get());
+
+    IdfExtensibleGroup eg2 = egs[1];
+    ASSERT_FALSE(eg2.getDouble(UnitarySystemPerformance_MultispeedExtensibleFields::HeatingSpeedSupplyAirFlowRatio));
+    EXPECT_EQ("autosize", eg2.getString(UnitarySystemPerformance_MultispeedExtensibleFields::HeatingSpeedSupplyAirFlowRatio).get());
+    ASSERT_TRUE(eg2.getDouble(UnitarySystemPerformance_MultispeedExtensibleFields::CoolingSpeedSupplyAirFlowRatio));
+    EXPECT_EQ(0.3, eg2.getDouble(UnitarySystemPerformance_MultispeedExtensibleFields::CoolingSpeedSupplyAirFlowRatio).get());
+  }
+}
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_UnitarySystemPerformanceMultispeed_AirLoopHVACUnitarySystem) {
```

Comment:
New test for translation of `UnitarySystemPerformanceMultispeed` using `AirLoopHVACUnitarySystem`.